### PR TITLE
Major refactoring and cleanup

### PR DESCRIPTION
**Pull request by j-adamczyk**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -23,9 +23,9 @@ python = "^3.9"
 datasketch = "*"
 e3fp = "*"
 joblib = "*"
-mhfp = "*"
 numpy = "^1.20.0"
 pandas = "*"
+pytest = "^8.1.1"
```
Comment (mjste) [inline pyproject.toml:28]: Why do we have pytest dependency added twice? (Second in test group dependencies)

Comment (j-adamczyk) [reply pyproject.toml]: Good catch, I removed it

**Review by mjste — CHANGES_REQUESTED** [review]
Tests are failing after last commit

**Comment by j-adamczyk** [comment]
@mjste fixed

**Pull request merged**